### ISSUE TITLE
Question about the training setting

### ISSUE THREAD

**Issue by zht022** [LiangSiyuan21/Adversarial-Attacks-for-Image-and-Video-Object-Detection#4]
Hi, @LiangSiyuan21 , thank you for sharing your training code. I've tried to train the GAN from scratch with this code but got confused. 

I've noticed that the implementation of your generator is not the same as you mentioned in the paper (encoder-decoder), while the discriminator is the same as [pix2pix] but not the ResNet-32 mentioned in the paper. 

Meanwhile, other parameters setting is almost the same as the paper: (1) objective function's weight \alpha=0.05, \beta=1, \epsilon=1e-4&2e-4; 
(2) use Adam solver with a learning rate of 0.0002 for D (but 0.0001 for G), and momentum parameters \beta1 = 0.5, \beta2 = 0.999; 
(3) divide the objective by 2 while optimizing D; 
(4) weights were initialized from a Gaussian distribution with mean 0 and standard deviation 0.02;
(5) for each input image, min_size & max_size = 300. 
Note that (3), (4), (5) are referred from [pix2pix].

However, after 3 training epoch, I've discovered that:
(1) The attacking result is not good enough. 
(2) The adv_images are strange. Distance between real & fake are not very closed, the objects in the adv_images are often with green shadows around them. Some examples are given here:

![image](https://user-images.githubusercontent.com/30546580/73326225-6f658400-428c-11ea-8c4a-2f47d30841ff.png)

![image](https://user-images.githubusercontent.com/30546580/73281361-d3eff700-422a-11ea-85d6-7c95bf13af7a.png)
![image](https://user-images.githubusercontent.com/30546580/73326151-2f9e9c80-428c-11ea-8670-bc89e1c9d0ac.png)
![image](https://user-images.githubusercontent.com/30546580/73326196-578e0000-428c-11ea-9d83-46c4d2d39f82.png)

My docker environment: python 3.7, pytorch 1.2.0, cuda 10.0, cudnn 7, cupy-cuda100.

If there exists any mistake in the above settings, please figure it out. What's more, I'm wondering how to check the correctness of the training process, if you can provide a training log, I would greatly appreciate!

**Comment by DuanXiaoyue-LittleMoon** [comment]
> 

Hello, I'm wondering whether you've meet with the problem that the misclassification loss in these 4 losses is always 0 during the training process and how you solved it? Thanks very much.
